extensions/amp-viewer-integration/0.1/messaging.js

```diff
@@ -66,123 +86,174 @@ export class Messaging {
   handleMessage_(event) {
     if (!event || event.source != this.target_ ||
       event.origin != this.targetOrigin_) {
+      this.logError_(TAG +
+        ': handleMessage_ failed. This message is not for us: ', event);
       return;
     }
+    /** @type {Message} */
     const message = event.data;
-    if (message.sentinel == requestSentinel_) {
+    if (message.app != APP) {
+      this.logError_(
+        TAG + ': handleMessage_ failed, wrong APP: ', event);
+      return;
+    }
+    if (message.type == MessageType.REQUEST) {
       this.handleRequest_(message);
-    } else if (message.sentinel == responseSentinel_) {
+    } else if (message.type == MessageType.RESPONSE) {
       this.handleResponse_(message);
-    } else {
-      throw new Error('Invalid Format!');
     }
   }
 
   /**
    * I'm sending Bob a new outgoing request.
-   * @param {string} eventType
-   * @param {*} payload
+   * @param {string} messageName
+   * @param {*} messageData
    * @param {boolean} awaitResponse
    * @return {!Promise<*>|undefined}
    */
-  sendRequest(eventType, payload, awaitResponse) {
-    dev().info(TAG, 'messaging.js -> sendRequest, eventType: ', eventType);
-    const requestId = String(++this.requestIdCounter_);
+  sendRequest(messageName, messageData, awaitResponse) {
+    dev().info(TAG, 'sendRequest, event name: ', messageName);
+    const requestId = ++this.requestIdCounter_;
     let promise = undefined;
     if (awaitResponse) {
       promise = new Promise((resolve, reject) => {
         this.waitingForResponse_[requestId] = {resolve, reject};
       });
     }
-    this.sendMessage_(requestSentinel_, requestId, eventType,
-      payload, awaitResponse);
+    this.sendMessage_({
+      app: APP,
+      requestid: requestId,
+      type: MessageType.REQUEST,
+      name: messageName,
+      data: messageData,
+      rsvp: awaitResponse,
+      error: undefined,
+    });
     return promise;
   }
 
   /**
    * I'm responding to a request that Bob made earlier.
-   * @param {string} requestId
-   * @param {*} payload
+   * @param {number} requestId
+   * @param {string} messageName
+   * @param {*} messageData
    * @private
    */
-  sendResponse_(requestId, payload) {
-    dev().info(TAG, 'messaging.js -> sendResponse_');
-    this.sendMessage_(
-      responseSentinel_, requestId, null, payload, false);
+  sendResponse_(requestId, messageName, messageData) {
+    dev().info(TAG, 'sendResponse_');
+    this.sendMessage_({
+      app: APP,
+      requestid: requestId,
+      type: MessageType.RESPONSE,
+      name: messageName,
+      data: messageData,
+      rsvp: undefined,
+      error: undefined,
+    });
   }
 
   /**
-   * @param {string} sentinel
-   * @param {string} requestId
-   * @param {string|null} eventType
-   * @param {*} payload
-   * @param {boolean} awaitResponse
+   * @param {number} requestId
+   * @param {string} messageName
+   * @param {string|undefined} reason
```

Comment:
Either it should be `!Error` here or the conversion should be done when calling this method.